src/blas/KokkosBlas3_gemm.hpp

```diff
@@ -199,15 +200,22 @@ gemm (const char transA[],
     }
   #endif // KOKKOSKERNELS_DEBUG_LEVEL > 0
 
-  // Return if degenerated matrices are provided
-  if((A.extent(0) == 0) || (A.extent(1) == 0) || (C.extent(1) == 0))
+  // Return if C matrix is degenerated
+  if(C.extent(1) == 0) {
     return;
+  }
+
+  // Simply scale C if A matrix is degenerated
+  if((A.extent(0) == 0) || (A.extent(1) == 0)) {
```

Comment:
``A.extent(0) == C.extent(0)`` so if we get here (assuming you change line 204), we know ``A.extent(0) != 0``. That way you would only need to check ``A.extent(1) == 0`` here.

Reply:
Yes, that makes sense, I will do both changes concurrently to update the logic, thanks for that good observation.